src/sage/matrix/matrix1.pyx

```diff
@@ -699,16 +706,20 @@ cdef class Matrix(Matrix0):
         Type ``numpy.typecodes`` for a list of the possible
         typecodes::
 
-            sage: import numpy
-            sage: sorted(numpy.typecodes.items())
-            [('All', '?bhilqpBHILQPefdgFDGSUVOMm'), ('AllFloat', 'efdgFDG'), ('AllInteger', 'bBhHiIlLqQpP'), ('Character', 'c'), ('Complex', 'FDG'), ('Datetime', 'Mm'), ('Float', 'efdg'), ('Integer', 'bhilqp'), ('UnsignedInteger', 'BHILQP')]
+            sage: import numpy                                                          # needs numpy
+            sage: sorted(numpy.typecodes.items())                                       # needs numpy
+            [('All', '?bhilqpBHILQPefdgFDGSUVOMm'), ('AllFloat', 'efdgFDG'),
+             ('AllInteger', 'bBhHiIlLqQpP'), ('Character', 'c'), ('Complex', 'FDG'),
+             ('Datetime', 'Mm'), ('Float', 'efdg'), ('Integer', 'bhilqp'),
+             ('UnsignedInteger', 'BHILQP')]
 
         Alternatively, numpy automatically calls this function (via
         the magic :meth:`__array__` method) to convert Sage matrices
         to numpy arrays::
 
+            sage: # needs numpy
             sage: import numpy
-            sage: b=numpy.array(a); b
+            sage: b = numpy.array(a); b
             array([[ 0,  1,  2,  3],
                    [ 4,  5,  6,  7],
                    [ 8,  9, 10, 11]])
```

Comment:
When I test this file, I have the following warning for this block. I don't understand why. Can we avoid this warning ?
```
sage -t --long --warn-long 20.5 --random-seed=138094706570993025836638675226273367820 src/sage/matrix/matrix1.pyx
**********************************************************************
File "src/sage/matrix/matrix1.pyx", line 729, in sage.matrix.matrix1.Matrix.numpy
Warning: Variable 'b' referenced here was set only in doctest marked '# needs numpy'
    b.shape
```

Reply:
Interesting. My guess is that this is caused by the `# 32-bit` / `# 64-bit` in this doctest. I'll investigate.

Reply:
Confirmed. That's now #36099

Reply:
OK. So I conclude the review.